chain/index/helpers.go

```diff
@@ -144,7 +144,7 @@ func withTx(ctx context.Context, db *sql.DB, fn func(*sql.Tx) error) error {
 }
 
 func isRetryableError(err error) bool {
-	return err != nil && strings.Contains(err.Error(), "database is locked")
+	return err != nil && (strings.Contains(err.Error(), "database is locked") || strings.Contains(err.Error(), "statement is closed"))
```

Comment:
I'm not convinced this is the right approach, why is the statement getting closed? it's not a network connection so shouldn't suffer from network problems. This seems like it's masking a real problem.

Reply:
@rvagg I see what you mean. Have reverted this change.